prov/efa/src/rdm/rxr_ep.c

```diff
@@ -427,6 +383,30 @@ struct rxr_op_entry *rxr_ep_alloc_tx_entry(struct rxr_ep *rxr_ep,
 	return tx_entry;
 }
 
+/**
+ * @brief convert EFA descriptors to shm descriptors.
+ *
+ * Each provider defines its descriptors format. The descriptor for
+ * EFA provider is of struct efa_mr *, which shm provider cannot
+ * understand. This function convert EFA descriptors to descriptors
+ * shm can use.
+ *
+ * @param numdesc[in]       number of descriptors in the array
+ * @param efa_desc[in]      efa descriptors
+ * @param shm_desc[out]     shm descriptors
+ *                          is shm descriptor.
+ */
+void rxr_get_desc_for_shm(int numdesc, void **efa_desc, void **shm_desc)
```

Comment:
Largely duplicates `rxr_convert_desc_for_shm` - can we merge them and only keep 1?

Reply:
That should be the case in my comment https://github.com/ofiwg/libfabric/pull/8708#discussion_r1149684783, which I think could involve more function call stacks.

Reply:
After this change the only remaining usage of `rxr_convert_desc_for_shm` is [here](https://github.com/ofiwg/libfabric/blob/2b33134370e1634c1612104865aa429eaadab102/prov/efa/src/rdm/rxr_atomic.c#L61)

I suggest we get rid of that and only keep `rxr_get_desc_for_shm`

Reply:
That can be doable. Will update.

Reply:
Latest revision uses `rxr_get_desc_for_shm` everywhere and gets rid of `rxr_convert_desc_for_shm`